systemtest/src/test/java/io/strimzi/systemtest/upgrade/kraft/KRaftStrimziDowngradeST.java

```diff
@@ -71,41 +70,47 @@ private void performDowngrade(BundleVersionModificationData downgradeData) throw
         // We support downgrade only when you didn't upgrade to new inter.broker.protocol.version and log.message.format.version
         // https://strimzi.io/docs/operators/latest/full/deploying.html#con-target-downgrade-version-str
 
-        setupEnvAndUpgradeClusterOperator(downgradeData, testStorage, testUpgradeKafkaVersion, TestConstants.CO_NAMESPACE);
-
-        logPodImages(TestConstants.CO_NAMESPACE);
+        setupEnvAndUpgradeClusterOperator(clusterOperatorNamespaceName, downgradeData, testStorage, testUpgradeKafkaVersion);
+        logClusterOperatorPodImage(clusterOperatorNamespaceName);
 
-        boolean wasUTOUsedBefore = StUtils.isUnidirectionalTopicOperatorUsed(TestConstants.CO_NAMESPACE, eoSelector);
+        boolean wasUTOUsedBefore = StUtils.isUnidirectionalTopicOperatorUsed(componentsNamespaceName, eoSelector);
 
         // Downgrade CO
-        changeClusterOperator(downgradeData, TestConstants.CO_NAMESPACE);
+        changeClusterOperator(clusterOperatorNamespaceName, componentsNamespaceName, downgradeData);
 
         // Wait for Kafka cluster rolling update
-        waitForKafkaClusterRollingUpdate();
-
-        logPodImages(TestConstants.CO_NAMESPACE);
+        waitForKafkaClusterRollingUpdate(componentsNamespaceName);
+        logComponentsPodImages(componentsNamespaceName);
 
         // Downgrade kafka
-        changeKafkaAndMetadataVersion(downgradeData);
+        changeKafkaAndMetadataVersion(componentsNamespaceName, downgradeData);
 
         // Verify that pods are stable
-        PodUtils.verifyThatRunningPodsAreStable(TestConstants.CO_NAMESPACE, clusterName);
+        PodUtils.verifyThatRunningPodsAreStable(componentsNamespaceName, clusterName);
 
-        checkAllImages(downgradeData, TestConstants.CO_NAMESPACE);
+        checkAllComponentsImages(componentsNamespaceName, downgradeData);
 
         // Verify upgrade
-        verifyProcedure(downgradeData, testStorage.getContinuousProducerName(), testStorage.getContinuousConsumerName(), TestConstants.CO_NAMESPACE, wasUTOUsedBefore);
+        verifyProcedure(componentsNamespaceName, downgradeData, testStorage.getContinuousProducerName(), testStorage.getContinuousConsumerName(), wasUTOUsedBefore);
     }
 
     @BeforeEach
     void setupEnvironment() {
         NamespaceManager.getInstance().createNamespaceAndPrepare(CO_NAMESPACE);
+
+        if (!CO_NAMESPACE.equals(TEST_SUITE_NAMESPACE)) {
+            NamespaceManager.getInstance().createNamespaceAndPrepare(TEST_SUITE_NAMESPACE);
+        }
```

Comment:
When this will happen?

Reply:
was only for safety meassures from me as i added this only additionally, but yeah good point this really seems unnecessary.